src/query/frontend/opencypher/grammar/MemgraphCypher.g4

```diff
@@ -364,14 +371,24 @@ setReplicationRole : SET REPLICATION ROLE TO ( MAIN | REPLICA )
 
 showReplicationRole : SHOW REPLICATION ROLE ;
 
-replicaName : symbolicName ;
+showReplicationCluster : SHOW REPLICATION CLUSTER ;
+
+instanceName : symbolicName ;
 
 socketAddress : literal ;
 
-registerReplica : REGISTER REPLICA replicaName ( SYNC | ASYNC )
+coordinatorSocketAddress : literal ;
+
+registerReplica : REGISTER REPLICA instanceName ( SYNC | ASYNC )
                 TO socketAddress ;
 
-dropReplica : DROP REPLICA replicaName ;
+registerReplicaCoordinatorServer: REGISTER REPLICA instanceName ( ASYNC | SYNC ) TO socketAddress WITH COORDINATOR SERVER ON coordinatorSocketAddress ;
+
+registerMainCoordinatorServer: REGISTER MAIN instanceName WITH COORDINATOR SERVER ON coordinatorSocketAddress ;
```

Comment:
In the current phase if the main comes back up it will not be handled in any case. This is something that might come in phase 3, but we need to check if we want to handle all that in manual failover, or we will push that to phase 4 when we start dealing with automatic failover. 


I am not sure @as51340 and me understand how can ASYNC/SYNC not be tied to instance regardless of its role, as SYNC/ASYNC is property of REPLICA. 

In the current phase, we decided not to have any communication from MAIN or REPLICAS to the coordinator, which means we need to have info on REPLICA sync mode to propagate that to the new main once it is down

Reply:
resolved on call